src/autoscaler/api/server/api_handler_test.go

```diff
@@ -0,0 +1,370 @@
+				Expect(resp.Code).To(Equal(http.StatusOK))
+			})
+		})
+
+		Context("When database DeleteServiceInstance call returns ErrDoesnotExist", func() {
+			BeforeEach(func() {
+				instanceDeletionRequestBody := &models.BrokerCommonRequestBody{
+					ServiceID: "a-service-id",
+					PlanID:    "a-plan-id",
+				}
+				body, err := json.Marshal(instanceDeletionRequestBody)
+				Expect(err).NotTo(HaveOccurred())
+				req, err = http.NewRequest(http.MethodPut, "", bytes.NewReader(body))
+
+				bindingdb.DeleteServiceInstanceReturns(db.ErrDoesNotExist)
+			})
+			It("fails with 410", func() {
+				Expect(resp.Code).To(Equal(http.StatusGone))
+			})
+		})
+
+		Context("When database DeleteServiceInstance call returns error other than ErrDoesnotExist", func() {
+			BeforeEach(func() {
+				instanceDeletionRequestBody := &models.BrokerCommonRequestBody{
+					ServiceID: "a-service-id",
+					PlanID:    "a-plan-id",
+				}
+				body, err := json.Marshal(instanceDeletionRequestBody)
+				Expect(err).NotTo(HaveOccurred())
+				req, err = http.NewRequest(http.MethodPut, "", bytes.NewReader(body))
+
+				bindingdb.DeleteServiceInstanceReturns(fmt.Errorf("error"))
+			})
+			It("fails with 500", func() {
+				Expect(resp.Code).To(Equal(http.StatusInternalServerError))
+			})
+		})
+
+	})
+
+	Describe("BindServiceInstance", func() {
+		JustBeforeEach(func() {
+			handler.BindServiceInstance(resp, req, map[string]string{"instanceId": "an-instance-id", "bindingId": "a-binding-id"})
+		})
+		Context("When mandatory parameters are not provided", func() {
+			BeforeEach(func() {
+				bindingRequestBody := &models.BrokerCommonRequestBody{
+					ServiceID: "a-service-id",
+				}
+				body, err := json.Marshal(bindingRequestBody)
+				Expect(err).NotTo(HaveOccurred())
+
+				req, err = http.NewRequest(http.MethodPut, "", bytes.NewReader(body))
+			})
+			It("fails with 400", func() {
+				Expect(resp.Code).To(Equal(http.StatusBadRequest))
+			})
+		})
+
+		Context("When mandatory parameters are present", func() {
+			BeforeEach(func() {
+				bindingRequestBody := &models.BrokerCommonRequestBody{
+					ServiceID: "a-service-id",
+					PlanID:    "a-plan-id",
+				}
+				body, err := json.Marshal(bindingRequestBody)
+				Expect(err).NotTo(HaveOccurred())
+
+				req, err = http.NewRequest(http.MethodPut, "", bytes.NewReader(body))
+			})
+			It("fails with 201", func() {
+				Expect(resp.Code).To(Equal(http.StatusCreated))
+			})
+		})
+
+		Context("When database CreateServiceBinding call returns ErrAlreadyExists", func() {
+			BeforeEach(func() {
+				bindingRequestBody := &models.BrokerCommonRequestBody{
+					ServiceID: "a-service-id",
+					PlanID:    "a-plan-id",
+				}
+				body, err := json.Marshal(bindingRequestBody)
+				Expect(err).NotTo(HaveOccurred())
+
+				req, err = http.NewRequest(http.MethodPut, "", bytes.NewReader(body))
+
+				bindingdb.CreateServiceBindingReturns(db.ErrAlreadyExists)
+			})
+			It("succeeds with 200", func() {
```

Comment:
The status code should be 409.

Reply:
> 200 SHOULD be returned if the binding already exists and the requested parameters are identical to the existing binding. The expected response body is below.
https://github.com/openservicebrokerapi/servicebroker/blob/v2.14/spec.md#binding